### PR TITLE
RD-2873 Accept blueprints as multipart/form

### PR DESCRIPTION
**Pull request by geokala**
This means we can stop messing around with dict encoding in query params.

### CODE REVIEW

**Comment by geokala** [comment]
This is part of the following batch of PRs:
https://github.com/cloudify-cosmo/cloudify-common/pull/1144
https://github.com/cloudify-cosmo/cloudify-manager/pull/3826
https://github.com/cloudify-cosmo/cloudify-cli/pull/1439
https://github.com/cloudify-cosmo/cloudify-stage/pull/2171

As noted in several places, it's to avoid us having to do silly things with encoding of query params (and thus make the new snapshots not require more ugliness).

**Review by tehasdf — DISMISSED** [review]

File: rest-service/manager_rest/test/mocks.py [hunk 1]
```diff
@@ -117,6 +103,23 @@ def _do_request(self, requests_method, request_url, body, params, headers,
...
+                        and body[entry][0] == 'filename'
+                    ):
+                        body[entry] = FileStorage(
+                            stream=open(body[entry][1].name, 'rb'))
+
+                # Remove application/json header to not override multipart
+                for header in headers:
+                    if headers[header] == 'application/json':
+                        break
+                headers.pop(header)
```
Comment (tehasdf) [inline rest-service/manager_rest/test/mocks.py:121]: this will always remove the last header. Write this differently, eg. using a dictcomp

Comment (geokala) [reply rest-service/manager_rest/test/mocks.py:121]: It won't, it'll remove the one that has the value of application/json- but yes, dictcomp would be better.

Comment (tehasdf) [reply rest-service/manager_rest/test/mocks.py:121]: and if there isn't one that has the value of application/json, then it'll remove an unrelated header that just happens to iterate last

Comment (tehasdf) [reply rest-service/manager_rest/test/mocks.py:121]: Also, that header's name is always content-type

Comment (geokala) [reply rest-service/manager_rest/test/mocks.py:121]: Ah, good point, yes- anyway, dictcomp version incoming.

Comment (geokala) [reply rest-service/manager_rest/test/mocks.py]: It's actually Content-type for the application/json. The one that gets added by the multipartencoder is Content-Type.

File: rest-service/manager_rest/test/endpoints/test_utils.py [hunk 2]
```diff
@@ -26,15 +26,17 @@ def generate_progress_func(total_size, buffer_size=8192):
     iteration = [0]
     max_iterations = total_size // buffer_size
 
-    def print_progress(read, total):
+    def print_progress(watcher):
+        read_bytes, total_bytes = watcher.bytes_read, watcher.len
+
         i = iteration[0]
-        assert total == total_size
+        assert read_bytes == total_bytes
```
Comment (tehasdf) [inline rest-service/manager_rest/test/endpoints/test_utils.py:33]: this one should be `total_size == total_bytes`

**Review by tehasdf — APPROVED** [review]
OK, because CI doesn't seem to work, but if it did, I'd have suggested the changes from #3831 first

**Pull request merged**